lib/backgroundscanner.php

```diff
@@ -51,58 +51,83 @@ public function run(){
 			return;
 		}
 		// locate files that are not checked yet
-		$dirMimetypeId = \OC::$server->getMimeTypeLoader()->getId('httpd/unix-directory');
-		$sql = 'SELECT `*PREFIX*filecache`.`fileid`, `*PREFIX*storages`.*'
-			.' FROM `*PREFIX*filecache`'
-			.' LEFT JOIN `*PREFIX*files_antivirus` ON `*PREFIX*files_antivirus`.`fileid` = `*PREFIX*filecache`.`fileid`'
-			.' JOIN `*PREFIX*storages` ON `*PREFIX*storages`.`numeric_id` = `*PREFIX*filecache`.`storage`'
-			.' WHERE `mimetype` != ?'
-			.' AND (`*PREFIX*storages`.`id` LIKE ? OR `*PREFIX*storages`.`id` LIKE ?)'
-			.' AND (`*PREFIX*files_antivirus`.`fileid` IS NULL OR `mtime` > `check_time`)'
-			.' AND `path` LIKE ?'
-			.' AND `*PREFIX*filecache`.`size` != 0';
-		$stmt = \OCP\DB::prepare($sql, 5);
+		$dirMimeTypeId = \OC::$server->getMimeTypeLoader()->getId('httpd/unix-directory');
 		try {
-			$result = $stmt->execute(array($dirMimetypeId, 'local::%', 'home::%', 'files/%'));
-			if (\OCP\DB::isError($result)) {
-				\OCP\Util::writeLog('files_antivirus', __METHOD__. 'DB error: ' . \OCP\DB::getErrorMessage($result), \OCP\Util::ERROR);
-				return;
-			}
+			$qb = \OC::$server->getDatabaseConnection()->getQueryBuilder();
+			$qb->select(['fc.fileid'])
+				->from('filecache', 'fc')
+				->leftJoin('fc', 'files_antivirus', 'fa', $qb->expr()->eq('fa.fileid', 'fc.fileid'))
+				->innerJoin(
+					'fc',
+					'storages',
+					'ss',
+					$qb->expr()->andX(
+						$qb->expr()->eq('fc.storage', 'ss.numeric_id'),
+						$qb->expr()->orX(
+							$qb->expr()->like('ss.id', $qb->expr()->literal('local::%')),
+							$qb->expr()->like('ss.id', $qb->expr()->literal('home::%'))
```

Comment:
note: this won't work for md5 storage ids https://github.com/owncloud/core/wiki/Storage-IDs#random-numberstring
Will need separate fixing.

The way how repair steps do is iterate over all users and get their home storages... yes, cumbersome.